api/src/main/java/com/xwiki/azureoauth/internal/AzureADObjectUpdateListener.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.azureoauth.internal;
+
+import java.util.Objects;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.phase.Initializable;
+import org.xwiki.component.phase.InitializationException;
+import org.xwiki.configuration.ConfigurationSaveException;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.model.reference.WikiReference;
+import org.xwiki.observation.AbstractEventListener;
+import org.xwiki.observation.event.Event;
+import org.xwiki.query.QueryException;
+import org.xwiki.stability.Unstable;
+import org.xwiki.wiki.descriptor.WikiDescriptorManager;
+
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.internal.event.XObjectUpdatedEvent;
+import com.xpn.xwiki.objects.BaseObjectReference;
+import com.xwiki.azureoauth.configuration.EntraIDConfiguration;
+
+import static com.xwiki.azureoauth.internal.configuration.EntraIDConfigurationSource.CONFIG_DOC;
+
+/**
+ * Checks the current installation version and transfers the old configuration from Identity OAuth to the new
+ * configuration class from OIDC. Listens to the modification made on the Azure AD configuration to update the OIDC
+ * client configuration.
+ *
+ * @version $Id$
+ * @since 2.0
+ */
+@Component
+@Named(AzureADObjectUpdateListener.HINT)
+@Singleton
+@Unstable
+public class AzureADObjectUpdateListener extends AbstractEventListener implements Initializable
```

Comment:
This could also use `EntraID` in the naming / description

I know it's annoying.. This change added a lot of confusion, I'm not sure why they did it (I think it's  troublesome for Microsoft too). But I feel like we should do our best to remove the `Azure` mentions whenever possible, and this big OIDC change it's the perfect opportunity to use `EntraID` in all new code